gcp-pubsub-http-connector/main.go

```diff
@@ -0,0 +1,183 @@
+		connectordata: connectordata,
+		logger:        logger,
+	}
+
+	logger.Sugar().Infof("Conn: %s", conn.connectordata.ResponseTopic)
+	conn.consumeMessage()
+}
+
+func (conn pubsubConnector) consumeMessage() {
+
+	creds := []byte(conn.pubsubInfo.Creds)
+	headers := http.Header{
+		"KEDA-Topic":          {conn.connectordata.Topic},
+		"KEDA-Response-Topic": {conn.connectordata.ResponseTopic},
+		"KEDA-Error-Topic":    {conn.connectordata.ErrorTopic},
+		"Content-Type":        {conn.connectordata.ContentType},
+		"KEDA-Source-Name":    {conn.connectordata.SourceName},
+	}
+	ctx := context.Background()
+	client, err := pubsub.NewClient(ctx, conn.pubsubInfo.ProjectID, option.WithCredentialsJSON(creds))
+	if err != nil {
+		conn.logger.Sugar().Errorf("pubsub.NewClient: %v", err)
+	}
+
+	var mu sync.Mutex
+	sub := client.Subscription(conn.pubsubInfo.SubscriptionID)
+	cctx, _ := context.WithCancel(ctx)
+
+	err = sub.Receive(cctx, func(ctx context.Context, msg *pubsub.Message) {
+		mu.Lock()
+		defer mu.Unlock()
+
+		// Set the attributes as message header came from PubSub record
+		for k, v := range msg.Attributes {
+			headers.Add(k, v)
+		}
+
+		// Push the message to the endpoint
+		resp, err := common.HandleHTTPRequest(string(msg.Data), headers, conn.connectordata, conn.logger)
+		if err != nil {
+			if conn.connectordata.ErrorTopic {
+				conn.responseOrErrorHandler(conn.connectordata.ErrorTopic, "Error", headers)
+			}
+			conn.logger.Error("Error sending the message to the endpoint %v", zap.Error(err))
+		} else {
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				if conn.connectordata.ErrorTopic {
+					conn.responseOrErrorHandler(conn.connectordata.ErrorTopic, string(body), headers)
+				}
+				conn.logger.Error("Error reading body", zap.Error(err))
+				resp.Body.Close()
+			} else {
+				msg.Ack()
+				if conn.connectordata.ResponseTopic {
+					conn.responseOrErrorHandler(conn.connectordata.ResponseTopic, string(body), headers)
+				}
+				conn.logger.Info("Success in sending the message", zap.Any("Messsage sent:  ", msg))
+			}
+		}
+
+	})
+
+}
+
+func (conn pubsubConnector) responseOrErrorHandler(topicID string, response string, headers http.Header) {
+
+	ctx := context.Background()
+	client, err := pubsub.NewClient(ctx, conn.pubsubInfo.ProjectID, option.WithCredentialsJSON([]byte(conn.pubsubInfo.Creds)))
+	if err != nil {
+		conn.logger.Error("pubsub.NewClient: %v", zap.Error(err))
+	}
+
+	t := client.Topic(topicID)
+	result := t.Publish(ctx, &pubsub.Message{
+		Data:       []byte(response),
+		Attributes: convHeadersToAttr(headers),
+	})
+
+	go func(res *pubsub.PublishResult) {
+		// The Get method blocks until a server-generated ID or
+		// an error is returned for the published message.
+		_, err := res.Get(ctx)
+		if err != nil {
+			conn.logger.Error("Failed to publish: %v", zap.Error(err))
+			return
+		}
+	}(result)
+	conn.logger.Sugar().Infof("Published message , topic name: %v\n", topicID)
```

Comment:
@RealHarshThakur Any particular reason to use Sugar() wrapper? Doc says it bit slower https://godoc.org/go.uber.org/zap#Logger.Sugar

Reply:
In other connectors, we have used key value format in the loggers.

Reply:
Fixed